SU2_CFD/include/solvers/CSpeciesSolver.hpp

```diff
@@ -153,6 +153,26 @@ class CSpeciesSolver : public CScalarSolver<CSpeciesVariable> {
    * (like BC_Isothermal_Wall) are implemented the respective CHeatSolver implementations can act as a  starting
    * point ---*/
 
+  /*!
+   * \brief Generic implementation of the isothermal wall also covering CHT cases,
+   * for which the wall temperature is given by GetConjugateHeatVariable.
+   */
+  void BC_Isothermal_Wall_Generic(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                                  CNumerics* visc_numerics, CConfig* config, unsigned short val_marker,
+                                  bool cht_mode = false);
+
+  /*!
+   * \brief Impose the scalar wall boundary condition.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] conv_numerics - Description of the numerical method.
+   * \param[in] visc_numerics - Description of the numerical method.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] val_marker - Surface marker where the boundary condition is applied.
+   */
+  void BC_Isothermal_Wall(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                          CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) override;
+
```

Comment:
I had to add this because scalar upwind for species transport in compressible flows showed a hard Dirichlet boundary condition of Y=0 on isothermal solid walls.

Reply:
I have an idea why we see those results in the walls, we can discuss it this Tuesday. 
![image](https://github.com/su2code/SU2/assets/98025159/a819e4a2-f8d3-4f7a-8092-b213738fe851)